lib/construct.gi

```diff
@@ -28,30 +28,46 @@ InstallMethod(CreateEmptyFile,[IsString],
     PrintTo(name,"");
 end);
 
+InstallMethod(InitPolymakeObject,[IsPolymakeObject],
+        function(poly)
+    local appstring;
+    # In polymake 4.1 this is required.
+    appstring:=Concatenation(
+                        "_application polytope\n",
+                        "_type Polytope\n\n"
+                  );
+    AppendToPolymakeObject(poly, appstring);
+    return poly;
+end);
+
 # create a (possibly empty) polygon. 
 # That is, a record just containing the name of a file to use with polymake.
 # If the file does not exist, it is created.
 ####################
 InstallMethod(CreatePolymakeObjectFromFile,[IsString],
         function(name)
-    local filename;
+    local filename, rtn;
+    rtn:=Objectify(PolymakeObject,rec(filename:=name, dir:=POLYMAKE_DATA_DIR));
     filename:=Filename(POLYMAKE_DATA_DIR,name);
     if not IsExistingFile(filename)
        then
         CreateEmptyFile(filename);
+        InitPolymakeObject(rtn);
     fi;
-    return Objectify(PolymakeObject,rec(filename:=name, dir:=POLYMAKE_DATA_DIR));
+    return rtn;
```

Comment:
The code in this method is not covered by a test (which of course is not your fault). Just out of curiosity, have you tested the new version in this method at all? If so, perhaps your test input could be turned into a regular test and added to the PR?

Reply:
No, I missed that. 

Adding a  test for the second method does mean duplicating pplane.poly since the tests modify their input files. Having to duplicate tests to test duplicate code is a step in the wrong direction, removing the duplicate code would be better.

Reply:
Commit f83abeea61a223a548706e3f9c25ca96cb67d333 fixes the duplicate code and adapts the tests.